analyzere/base_resources.py

```diff
@@ -258,11 +263,16 @@ def upload_data(self, file_or_str):
             request_raw('post', self._data_path)
 
         # Upload chunks
-        for chunk, offset in utils.read_in_chunks(file_obj,
-                                                  analyzere.upload_chunk_size):
+        for chunk, offset in utils.read_in_chunks(file_obj, chunk_size):
             headers = {'Offset': str(offset),
                        'Content-Type': 'application/offset+octet-stream'}
             request_raw('patch', self._data_path, headers=headers, body=chunk)
+            # if there is a known size, and an upload callback, call it
+            if length and upload_callback:
+                upload_callback((offset/length) * 100.0)
+
+        if upload_callback:
+            upload_callback(100.0)
```

Comment:
an alternative approach, which I think makes everything even simpler is to do something like
```
noop = lambda x: None
upload_callback = upload_callback if callable(upload_callback) else noop
commit_callback = commit_callback if callable(commit_callback) else noop
```